sonar-xml-plugin/src/main/java/org/sonar/plugins/xml/highlighting/XMLHighlighting.java

```diff
@@ -73,14 +73,16 @@ private void highlightElementNode(Node node) {
     XmlTextRange lastLocation = nameLocation;
     for (int i = 0; i < attributes.getLength(); i++) {
       Attr attribute = (Attr) attributes.item(i);
-      addHighlighting(NewXmlFile.attributeNameLocation(attribute), TypeOfText.CONSTANT);
-      XmlTextRange valueLocation = NewXmlFile.attributeValueLocation(attribute);
+      addHighlighting(XmlFile.attributeNameLocation(attribute), TypeOfText.CONSTANT);
+      XmlTextRange valueLocation = XmlFile.attributeValueLocation(attribute);
       addHighlighting(valueLocation, TypeOfText.STRING);
       lastLocation = valueLocation;
     }
 
-    if (startLocation.equals(endLocation)) {
-      // self-closing element <foo ... />
+    // self-closing element <foo ... />
+    boolean isSelfClosing = startLocation.getEndLine() == endLocation.getEndLine() && startLocation.getEndColumn() == endLocation.getEndColumn();
```

Comment:
To be honest I think it would have made more sense to keep the equals method in the XmlTextRange